chef/cookbooks/crowbar/recipes/stop-services-before-upgrade.rb

```diff
@@ -33,26 +34,11 @@
   only_if { ::File.exist?("/usr/sbin/crm") }
 end
 
-# we deal with rabbitmq differently because of drbd; here we want to be more
-# careful about how we stop things as experience showed that it's easy to get
-# fenced if we're stopping things blindly
-bash "stop pacemaker resources for rabbitmq" do
-  code <<-EOF
-    for resource in g-rabbitmq rabbitmq fs-rabbitmq ms-drbd-rabbitmq drbd-rabbitmq;
-    do
-      if crm configure show $resource >/dev/null 2>&1; then
-        crm --force --wait resource stop $resource
-      fi
-    done
-  EOF
-  only_if { ::File.exist?("/usr/sbin/crm") }
-end
-
 # Stop openstack services on this node.
 # Note that for HA setup, they should be already stopped by pacemaker.
 bash "stop OpenStack services" do
   code <<-EOF
-    for i in /etc/init.d/openstack-* \
+    for i in `ls /etc/init.d/openstack-* | grep -Ev 'keystone|neutron'` \
```

Comment:
The idea was to have the L3 agents live as long as possible in the non-HA case. But now that you mention it, during my investigations you could stop all services and the agents were still alive. I'll check that again and see if we can remove this here.